src/nvim/ex_docmd.c

```diff
@@ -6329,6 +6329,13 @@ static void ex_quit(exarg_T *eap)
     }
     not_exiting();
     // close window; may free buffer
+    // close floats anchored to it first
```

Comment:
Ideally we we would check first if there are unrelated floats and abort the operation before closing any float, also in this case. Could break out the check to a helper function to avoid duplicating the code.

Reply:
I added the check for unanchored floats to the `win_close()` function itself. I guess adding it to a helper function makes more sense, I'll place a call to it in `ex_win_close()` and everywhere else before `win_close()` is called.

Reply:
@bfredl If i call the checker outside of `win_close()`, it won't be able to return `FAIL` if the window is not closed due to unanchored floats.

Reply:
What I said was that the check could be moved to a helper function to avoid duplicating code. I certainly did not say the check must forcefully removed from `win_close()` even if the check is needed in `win_close()`.

Reply:
@bfedl, I see, I misread the first comment. Sorry for that! :sweat_smile:

Reply:
Done!